### PR TITLE
Decode #13

### PR DESCRIPTION
**Pull request by nelsonic**
This PR: 
+ [x] Adds the `decode/1` function #13 
+ [x] Adds guard clauses for `integer`, `float` and `atom` values. #14 
+ [x] Enhanced the `README.md`: https://github.com/dwyl/base58/tree/decode-issue%2313

It can only be merged once we've looked at the the Module name; currently `Base58Encode` #16 
Publishing a new version is dependent on #15

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab please see: https://github.com/dwyl/base58/tree/decode-issue%2313

File: test/base58_test.exs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+defmodule Base58Test do
```
Comment (nelsonic) [inline test/base58_test.exs:1]: Don't know why GitHub thinks this is a new file, I just renamed it.
`.then` added a few new tests. 📝

File: mix.exs [hunk 2]
```diff
@@ -1,13 +1,13 @@
-defmodule Base58Encode.MixProject do
+defmodule Base58.MixProject do
```
Comment (nelsonic) [inline mix.exs:1]: Rename #16

File: lib/base58.ex [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  ## Examples
+
+    iex> Base58.encode("hello") |> Base58.decode()
+    "hello"
+  """
+  @alnum ~c(123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz)
+
+  def decode(""), do: "" # return empty string unmodified
+  def decode("\0"), do: "" # treat null values as empty
+  def decode(encoded), do: recurse(encoded |> to_char_list, 0)
```
Comment (SimonLab) [inline lib/base58.ex:97]: `to_char_list` is now deprecated and can be replaced by `to_charlist`:
```suggestion
  def decode(encoded), do: recurse(encoded |> to_charlist, 0)
```

File: example/example.exs [hunk 4]
```diff
@@ -1,8 +1,10 @@
 defmodule Example do
 
   hello = "hello"
-  helloBase58 = Base58Encode.encode(hello)
+  helloBase58 = Base58.encode(hello)
```
Comment (nelsonic) [inline example/example.exs:4]: @SimonLab thanks for updating the example. 👍

**Comment by nelsonic** [comment]
@SimonLab thanks for updating the example. 👍 
I cannot approve/merge my own PR:
image

Please merge if you're happy with it. 
merci bowcups. 😉

**Pull request merged**